pycbc/events/eventmgr_cython.pyx

```diff
@@ -118,6 +120,98 @@ def logsignalrateinternals_compute2detrate(
         rate[ridx] *= (rescale_fac*rescale_fac*rescale_fac*rescale_fac)
 
 
+@boundscheck(False)
+@wraparound(False)
+@cdivision(True)
+def timecoincidence_constructfold(
+    double[:] fold1,
+    double[:] fold2,
+    double[:] t1,
+    double[:] t2,
+    double slide_step,
+    int length1,
+    int length2
+):
+    cdef:
+        int idx
+
+    for idx in range(length1):
+        fold1[idx] = t1[idx] % slide_step
+
+    for idx in range(length2):
+        fold2[idx] = t2[idx] % slide_step
+
+
+@boundscheck(False)
+@wraparound(False)
+@cdivision(True)
+def timecoincidence_findidxlen(
+    long int[:] left,
+    long int[:] right,
+    int leftlength,
+):
+    cdef:
+        int idx, tslength
+
+    tslength = 0
+    for idx in range(leftlength):
+        tslength += right[idx] - left[idx]
+    return tslength
+
+
+@boundscheck(False)
+@wraparound(False)
+@cdivision(True)
+def timecoincidence_constructidxs(
+    unsigned int[:] idx1,
+    unsigned int[:] idx2,
+    long int[:] sort1,
+    long int[:] sort2,
+    long int[:] left,
+    long int[:] right,
+    int leftlength,
+    int sort2length
+):
+    cdef:
+        int idx, jdx, count, currlen
+
+    # Construct sort1
+    count = 0
+    for idx in range(leftlength):
+        currlen = right[idx] - left[idx]
+        for jdx in range(currlen):
+            idx1[count] = sort1[idx]
+            count += 1
+
+    # Construct sort2
+    count = 0
+    for idx in range(leftlength):
+        for jdx in range(left[idx], right[idx]):
+            idx2[count] = sort2[jdx % sort2length]
+            count += 1
+
+
+@boundscheck(False)
+@wraparound(False)
+@cdivision(True)
+def timecoincidence_getslideint(
+    int [:] slide,
+    double[:] t1,
+    double[:] t2,
+    unsigned int[:] idx1,
+    unsigned int[:] idx2,
+    double slide_step
+):
+    cdef:
+        int idx, length
+
+    length = idx1.shape[0]
+
+    for idx in range(length):
+        diff = (t1[idx1[idx]] / slide_step) - (t2[idx2[idx]] / slide_step)
```

Comment:
Would it be worth doing this as
```python
        diff = (t1[idx1[idx]] - t2[idx2[idx]]) / slide_step
```
and save a division?

Reply:
This *should* be optimized away in compilation, but worth avoiding this anyway, so this is done.